control/ipmitool.spec.in

```diff
@@ -34,7 +34,7 @@ fi
 %setup
 
 %build
-./configure --with-kerneldir \
```

Comment:
This was added with commit b85e953ab9dd75905796772e3f10b4d479d9b2e9
and is probably related to this commit log part: "add hooks for building rpm in arbitrary root".

I'm not very skilled in RPM crafting, so I can't tell if it is actually needed.
Can you confirm that removing this doesn't break anything?

Reply:
Well, the only thing I'm sure is that --with-kerneldir breaks the build at least with base Fedora 31 and that removing --with-kerneldir  makes "make rpm" build with exactly the same gcc flags than "make" which seems sane to me.
I don't think building a user-land code with kernel tree headers is a good idea nor safe.. Or at most "user" kernel headers under /usr/include/linux/ (from package kernel-headers) if needed.
I also don't see  what "add hooks for building rpm in arbitrary root" means.

Reply:
I think I understand the meaning of "add hooks for building rpm in arbitrary root".  It looks like a bad workaround for the case where the headers are not located at the right place when building the RPM and then not found at the expected relative paths. In this case that should never occur, the workaround takes the kernel tree headers at an absolute path.
With --with-kerneldir, the build flags are :
gcc -DHAVE_CONFIG_H -I. -I..  -I../include   -g -O2 -Wall -Wextra -std=gnu11 -pedantic -Wformat -Wformat-nonliteral -I /lib/modules/5.3.7-301.fc31.x86_64/build/include -MT ipmi_sel.lo -MD -MP -MF $depbase.Tpo -c -o ipmi_sel.lo ipmi_sel.c
where /lib/modules/5.3.7-301.fc31.x86_64/build/include is the only absolute path.